lib/puppet/node/environment.rb

```diff
@@ -82,8 +82,12 @@ def self.remote(name)
   # @param name [Symbol] The environment name
   def initialize(name, modulepath, manifest, config_version)
     @name = name
-    @modulepath = modulepath
-    @manifest = manifest
+    @modulepath = modulepath.map { |p| Puppet::FileSystem.expanded_path_string(p) }
+    if manifest && !manifest.is_a?(Symbol) && Puppet::FileSystem.exist?(manifest)
+      @manifest = Puppet::FileSystem.expanded_path_string(manifest)
+    else
+      @manifest = manifest
+    end
```

Comment:
One thing I need to understand better @hlindberg is the usage patterns here... is it necessary to create an environment to get instances of `module` and `module_loaders`?

If so, it probably just makes sense to do this work here and have it fan out as I mentioned.  But if those other instances are used stand-alone, outside of an environment... then that approach won't always work.